transformer_engine/common/recipe/__init__.py

```diff
@@ -7,9 +7,14 @@
 import warnings
 import os
 from enum import Enum
-from typing import Literal, Optional, Union, Callable, NamedTuple
+from typing import Literal, Optional, Type, Union, Callable, NamedTuple
 from pydantic.dataclasses import dataclass
 
+from transformer_engine.pytorch.tensor.float8_blockwise_tensor import Float8BlockwiseQTensor
+from transformer_engine.pytorch.tensor.float8_tensor import Float8Tensor
+from transformer_engine.pytorch.tensor.mxfp8_tensor import MXFP8Tensor
+from transformer_engine.pytorch.tensor.quantized_tensor import QuantizedTensor
```

Comment:
Actually you can't do it that way - this recipe class is in common, so is used in the JAX version as well. You will need to do the import and setting in the `__post_init__` method as well (with try/except block).

Reply:
Right, fixed in https://github.com/NVIDIA/TransformerEngine/pull/1724/commits/d6ea98167c39da0185aa89d2a98b080a1a26dfdb